src/zenml/models/v2/base/filter.py

```diff
@@ -380,6 +385,67 @@ def sorting_params(self) -> Tuple[str, SorterOps]:
 
         return column, operator
 
+    def configure_rbac(
+        self,
+        authenticated_user_id: UUID,
+        **column_allowed_ids: Optional[Set[UUID]],
+    ) -> None:
+        """Configure RBAC allowed column values.
+
+        Args:
+            authenticated_user_id: ID of the authenticated user. All entities
+                owned by this user will be included.
+            column_allowed_ids: Set of IDs per column to limit the query to.
+                If given, the remaining filters will be applied to entities
+                within this set only. If `None`, the remaining filters will
+                applied to all entries in the table.
+        """
+        self._rbac_configuration = (authenticated_user_id, column_allowed_ids)
+
+    def apply_rbac_filter(
+        self,
+        query: Union["Select[AnySchema]", "SelectOfScalar[AnySchema]"],
+        table: Type["AnySchema"],
+    ) -> Union["Select[AnySchema]", "SelectOfScalar[AnySchema]"]:
+        """Applies the RBAC filter to a query.
+
+        Args:
+            query: The query to which to apply the filter.
+            table: The query table.
+
+        Returns:
+            The query with RBAC filter applied.
+        """
+        from sqlmodel import or_
+
+        if not self._rbac_configuration:
+            return query
+
+        expressions = []
+
+        for column_name, allowed_ids in self._rbac_configuration[1].items():
+            if allowed_ids is not None:
+                expression = getattr(table, column_name).in_(allowed_ids)
```

Comment:
I probably mentioned this before: this query can explode if a lot of UUIDs are passed in. I think you should put a limit on how many IDs can be passed in a single column `IN` query like this, either here or wherever this is called (in the RBAC utilities, or the RBAC implementation ?) and raise an error. If the limit is enforced at the source (e.g. whenever someone adds an ID to an existing resource type for a role), then we can prevent performance degradation on this side of things.

There are some ways to implement large `IN` queries without degrading performance, like creating temporary entries in table, but perhaps we can avoid that altogether.

Reply:
What would you say is a reasonable limit for this? I guess we could also limit the amount of ids per role, but assigning multiple roles for a user would again allow them to exceed this limit?

Reply:
at a maximum, we could limit 100 IDs per role and 100 roles per user, this way we don't exceed 1000 IDs per query

Reply:
There's at least one `0` missing or too much for this to make sense 🤔

Reply:
I like the idea though, nicer than limiting it here when applying it as that would just lead to wrong results or error which could have been avoided at role creation/assignment time.

Reply:
Right, a zero missing :smiley: 50 IDs per role and 20 roles per user should be enough, I think

Reply:
Alright, I'll add such a limitation in the cloud backend

Reply:
Doesn't need to be done in this PR though. Can be a follow-up.

Reply:
The `teams` we have in the cloud poses some additional questions: Do we include the roles assigned via teams in this limit?
- If yes, we'd have to check the role limit for each user of a team whenever we assign a role to the team. This also means single users of a team might block assigning a role to a team.
- If no, people could avoid the role limit by creating an arbitrary amount of teams and using those to assign the roles.